### PR TITLE
Pass additional options into the makeServer function

### PR DESCRIPTION
**Pull request by cah-brian-gantzler**

### CODE REVIEW

File: packages/ember-mirage/addon/setup-mirage.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ import startMirage from './start-mirage';
  NOTE: the `hooks = self` is for mocha support
  @hide
  */
-export default function setupMirage(hooks = self, options) {
+export default function setupMirage(hooks = self, {makeServer, ...options}) {
```
Comment (cah-brian-gantzler) [inline packages/ember-mirage/addon/setup-mirage.js:14]: There was no reason to pass makeServer on, removed it from options

File: packages/ember-mirage/addon/setup-mirage.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ export default function setupMirage(hooks = self, options) {
       );
     }
 
-    this.server = startMirage(options.makeServer, { owner: this.owner, ...options });
+    this.server = startMirage(makeServer, { owner: this.owner, ...options });
```
Comment (cah-brian-gantzler) [inline packages/ember-mirage/addon/setup-mirage.js:24]: using that makeServer that was removed from options

File: packages/ember-mirage/addon/start-mirage.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { pluralize, singularize } from 'ember-inflector';
 
 import assert from './assert';
 
-export default function startMirage(makeServer, { owner, env } = {}) {
+export default function startMirage(makeServer, { owner, env, ...otherOptions } = {}) {
```
Comment (cah-brian-gantzler) [inline packages/ember-mirage/addon/start-mirage.js:5]: any additional options that the user might want to pass on were not being passed on.

File: packages/ember-mirage/addon/start-mirage.js [hunk 4]
```diff
@@ -21,6 +21,7 @@ export default function startMirage(makeServer, { owner, env } = {}) {
     env,
     environment,
     inflector: { singularize, pluralize },
+    ...otherOptions
```
Comment (cah-brian-gantzler) [inline packages/ember-mirage/addon/start-mirage.js:24]: pass those other options on

**Comment by cah-brian-gantzler** [comment]
I use husky to setup an auto run of prettier when a commit happens on all my work repos. So always forget to run it on open source repos that require it manually. Mind if I add Husky and run prettier and lint on commit?

**Review by NullVoxPopuli — APPROVED** [review]
very nice!

**Comment by NullVoxPopuli** [comment]
> Mind if I add Husky and run prettier and lint on commit?

as long as it's optional / doesn't hook in to `prepare`, that's fine.
The thing with husky is it requires you to have successfully installed your dependencies before it'll work -- which, when it blocks commits, can be really annoying for contributors who just want to change documentation (often the most neglected, and where the most help is needed)

**Comment by cah-brian-gantzler** [comment]
Husky hooks into git commit, so I think would be same thing. :( I always forget to run prettier lint manually and then test CI fails because it wasnt run. sigh

**Pull request merged**